test/test_ops.py

```diff
@@ -1102,6 +1104,27 @@ def test_vjpvmap(self, device, dtype, op):
 
                 self.assertEqual(result_vjps, expected_vjps)
 
+    def _compare_jacobians(self, fn, cotangents_and_primals, argnums=None, atol_rtol=None):
+        if argnums is None:
+            argnums = tuple(range(len(cotangents_and_primals)))
+
+        def get_vjp(cotangents, *primals):
+            _, vjp_fn = vjp(fn, *primals)
+            return vjp_fn(cotangents)
+
+        jacobian_jvp = jacfwd(get_vjp, argnums)(*cotangents_and_primals)
+        jacobian_vjp = jacrev(get_vjp, argnums)(*cotangents_and_primals)
+
+        # For dtype changing operations, the jacobians have different dtype.
```

Comment:
nit: would be nice to check that the jacobian is of the correct dtype, but I'm not sure if it is worth it. It seems complicated to do:

If the operation has inputs of dtype A and outputs of dtype B:
- then the vjp is a function that takes inputs of dtype (A, B) and gives outputs of dtype B
- so the jacobian_vjp should return jacobians of dtype A (for the primals) and dtype B (for the cotangents)
- the jacobian_jvp should return jacobians of dtype B

Reply:
> would be nice to check that the jacobian is of the correct dtype

The other difficult thing is that none of the decompositions we're testing change dtype with the inputs we're using. In other words, currently all of the `jvpvjp` tests and these new tests pass without these lines. I'll file an issue that we should see if we can add some tests that change dtypes and see if we can update this section with that